### ISSUE TITLE
map_and_batch() is way faster than map().batch() ???

### ISSUE THREAD

**Issue by hopehealth** [tensorflow/tensorflow#43489]
<em>Please make sure that this is an issue related to performance of TensorFlow.
As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:performance_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): tensorflow-gpu 2.2.0
- Python version: Python 3.5.2
- Bazel version (if compiling from source): None
- GCC/Compiler version (if compiling from source): gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
- CUDA/cuDNN version:  CUDA Version: 10.1 , libcudnn7-dev 7.6.5.32-1
- GPU model and memory: NVIDIA Tesla V100, Memory 32480MiB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
When training ResNet-50 with ImageNet (in TFRecord), the time per epoch with "map_and_batch()" is 3 times faster than using ".map().batch()".
I know "map_and_batch()" will be depracated (https://www.tensorflow.org/api_docs/python/tf/data/experimental/map_and_batch?hl=en). 
But if this is faster and working just fine, then why ??

**Describe the expected behavior**
Accoridng to some previous posts, these two API should have similar performance.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
On the same machine, when i use 
dataset = dataset.apply(tf.data.experimental.map_and_batch(
    lambda value: parse_record_fn(value, is_training, dtype),
    batch_size=batch_size))
the results are:
Epoch 1/30
2859/2859 [==============================] - 2638s 923ms/step - accuracy: 0.5294 - loss: 2.9772 - val_accuracy: 0.6064 - lr: 0.1000 - val_loss: 2.6385
Epoch 2/30
1700/2859 [================>.............] - ETA: 17:26 - accuracy: 0.6093 - loss: 2.6254

However, when we use
dataset = dataset.map(lambda value: parse_record_fn(value, is_training, dtype))
dataset = dataset.batch(batch_size=batch_size)
the results are:
Epoch 1/30
2859/2859 [==============================] - 6732s 2s/step - loss: 2.9723 - accuracy: 0.5304 - lr: 0.1000 - val_loss: 2.6656 - val_accuracy: 0.5948
Epoch 2/30
2110/2859 [=====================>........] - ETA: 27:56 - loss: 2.6110 - accuracy: 0.6128

**Comment by bhack** [comment]
Take a look at https://github.com/tensorflow/tensorflow/issues/20059
